### ISSUE TITLE
Decorator library adds spurious lines

### ISSUE THREAD

**Issue by alexmojaki** [cool-RR/PySnooper#109]
Example script:

```python
import pysnooper

@pysnooper.snoop(depth=2)
def factorial(x):
    if x <= 1:
        return 1
    return factorial(x - 1)

factorial(2)
```

Output:

```
Starting var:.. x = 2
18:12:31.470110 call         5 def factorial(x):
18:12:31.470305 line         6     if x <= 1:
18:12:31.470348 line         8     return factorial(x - 1)
    Starting var:.. x = 1
    18:12:31.470411 call         1 SOURCE IS UNAVAILABLE
    18:12:31.470513 line         2 SOURCE IS UNAVAILABLE
Starting var:.. x = 1
18:12:31.470591 call         5 def factorial(x):
18:12:31.470634 line         6     if x <= 1:
18:12:31.470671 line         7         return 1
18:12:31.470706 return       7         return 1
Return value:.. 1
    18:12:31.470774 return       2 SOURCE IS UNAVAILABLE
    Return value:.. 1
18:12:31.470824 return       8     return factorial(x - 1)
Return value:.. 1
```

When it tries to read the source it gets an error:

    [Errno 2] No such file or directory: '<decorator-gen-1>'

Obviously not being able to read the source is bad, but even if it could, it looks like we would get various unwanted lines, as any wrapper function is going to pollute the stack.

**Comment by cool-RR** [comment]
Do you prefer that we switch to `functools.wraps` and get rid of the decorator module?

**Comment by alexmojaki** [comment]
Yes. I haven't looked into the decorator module, but if all it does is preserve signatures, that doesn't seem very useful for this library.

**Comment by cool-RR** [comment]
Okay, feel free to remove it from the repo and switch to `functools.wraps`. Yay, one less bundled third-party module!

**Comment by alexmojaki** [comment]
Fixed in #112
